File: snake-strategy-bono/src/main/java/co/electric/snake/strategy/bonostrategy/newdirectionprocessor/ByKispalEsABorz.java
package co.electric.snake.strategy.bonostrategy.newdirectionprocessor;

import co.electric.snake.framework.model.Direction;
import co.electric.snake.strategy.bonostrategy.BlockingDirectionContainer;
import co.electric.snake.strategy.bonostrategy.SimpleDirectionContainer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ByKispalEsABorz implements NewDirectionProcessor {

    private static final int ORDER = 5;

    private static final Logger LOG = LoggerFactory.getLogger(ByKispalEsABorz.class);

    @Override
    public Direction process(SimpleDirectionContainer filteredDirections, SimpleDirectionContainer equivalentBestDirections, BlockingDirectionContainer blockingDirections) {
        Direction newDirection;

        SimpleDirectionContainer kispalDirections = new SimpleDirectionContainer();
        kispalDirections.add(Direction.SOUTH);

        newDirection = processFinalDirection(kispalDirections, LOG);

        return newDirection;
    }

    @Override
    public int getOrder() {
        return ORDER;
    }

}

File: snake-strategy-bono/src/main/kotlin/co/electric/snake/strategy/bonostrategy/newdirectionprocessor/ByKispalEsABorz.kt
package co.electric.snake.strategy.bonostrategy.newdirectionprocessor

import co.electric.snake.framework.model.Direction
import co.electric.snake.strategy.bonostrategy.BlockingDirectionContainer
import co.electric.snake.strategy.bonostrategy.SimpleDirectionContainer
import org.slf4j.LoggerFactory
import java.util.*

class ByKispalEsABorz : NewDirectionProcessor {

    companion object {
        private val LOG = LoggerFactory.getLogger(ByKispalEsABorz::class.java)
    }

    override val order = 6

    override fun process(filteredDirections: SimpleDirectionContainer, equivalentBestDirections: SimpleDirectionContainer, blockingDirections: BlockingDirectionContainer): Optional<Direction> {
        val kispalDirections = SimpleDirectionContainer()
        kispalDirections.add(Direction.SOUTH)
        return processFinalDirection(kispalDirections, LOG)
    }

}
